autowisp/browser_interface/configuration/views.py

```diff
@@ -31,10 +31,65 @@
 
 # pylint: enable=no-name-in-module
 
+def _merge_children(old_children, new_children, parent_type=None):
+    """Merge lists of child nodes by (name,type); keep order, append new."""
+    old_children = old_children or []
+    new_children = new_children or []
+
+    # For parameter nodes the child is usually a single 'value' node.
+    if parent_type == "parameter":
+        # If the new file supplied a value, use it; otherwise keep old.
+        return new_children if new_children else old_children
+
+    # Generic case: merge by (name, type)
+    merged = list(old_children)
+    index = {
+        (c.get("name"), c.get("type")): i
+        for i, c in enumerate(old_children)
+        if isinstance(c, dict)
+    }
+
+    for child in new_children:
+        if isinstance(child, dict):
+            key = (child.get("name"), child.get("type"))
+            if key in index:
+                i = index[key]
+                merged[i] = deep_merge_config(merged[i], child)
+                continue
+        merged.append(child)
+    return merged
+
+
+def deep_merge_config(existing, new):
+    """Recursively merge config trees; new overrides only where provided."""
+    if new is None:
+        return existing
+
+    if isinstance(existing, dict) and isinstance(new, dict):
```

Comment:
This should be an assert. We always expect the top level of the configuration to be a dict.